Ical.Net/DataTypes/CalDateTime.cs

```diff
@@ -174,271 +237,301 @@ public override int GetHashCode()
         => !(left == right);
 
-    public static TimeSpan operator -(CalDateTime left, IDateTime right)
-    {
-        left.AssociateWith(right);
-        return left.AsUtc - right.AsUtc;
-    }
-
+    /// <summary>
+    /// Subtracts a <see cref="TimeSpan"/> from the <see cref="CalDateTime"/>.
+    /// </summary>
+    /// <remarks>
+    /// This will also set <seealso cref="HasTime"/> to <see langword="true"/>,
+    /// if the <see cref="TimeSpan"/> is not a multiple of 24 hours.
+    /// </remarks>
     public static IDateTime operator -(CalDateTime left, TimeSpan right)
     {
-        var copy = left.Copy<IDateTime>();
+        var copy = left.Copy<CalDateTime>();
+        if (Math.Abs(right.TotalDays % 1) > AlmostZeroEpsilon)
+        {
+            copy.HasTime = true;
+        }
         copy.Value -= right;
         return copy;
     }
 
+    /// <summary>
+    /// Adds a <see cref="TimeSpan"/> to the <see cref="CalDateTime"/>.
+    /// </summary>
+    /// <remarks>
+    /// This will also set <seealso cref="HasTime"/> to <see langword="true"/>,
+    /// if the <see cref="TimeSpan"/> is not a multiple of 24 hours.
+    /// </remarks>
     public static IDateTime operator +(CalDateTime left, TimeSpan right)
     {
-        var copy = left.Copy<IDateTime>();
+        var copy = left.Copy<CalDateTime>();
+        if (Math.Abs(right.TotalDays % 1) > AlmostZeroEpsilon)
+        {
+            copy.HasTime = true;
+        }
         copy.Value += right;
         return copy;
     }
 
+    /// <summary>
+    /// Creates a new instance of <see cref="CalDateTime"/> with <see langword="true"/> for <see cref="HasTime"/>
+    /// </summary>
     public static implicit operator CalDateTime(DateTime left) => new CalDateTime(left);
 
     /// <summary>
-    /// Converts the date/time to the date/time of the computer running the program. If the DateTimeKind is Unspecified, it's assumed that the underlying
+    /// Converts the date/time to the date/time of the computer running the program.
+    /// If the DateTimeKind is Unspecified, it's assumed that the underlying
     /// Value already represents the system's datetime.
     /// </summary>
-    public DateTime AsSystemLocal
-    {
-        get
-        {
-            if (Value.Kind == DateTimeKind.Unspecified)
-            {
-                return HasTime
-                    ? Value
-                    : Value.Date;
-            }
+    public DateTime AsSystemLocal => AsDateTimeOffset.LocalDateTime;
 
-            return HasTime
-                ? Value.ToLocalTime()
-                : Value.ToLocalTime().Date;
-        }
-    }
+    /// <summary>
+    /// Returns a representation of the <see cref="DateTime"/> in UTC.
+    /// </summary>
+    public DateTime AsUtc => AsDateTimeOffset.UtcDateTime;
+
+    /// <summary>
+    /// Gets the underlying <see cref="DateOnlyValue"/> of <see cref="Value"/>.
+    /// </summary>
+    public DateOnly? DateOnlyValue => HasDate ? _dateOnly : null;
```

Comment:
Maybe just name it `Date`, in replacement of the existing, now obsolete `Date`?

Reply:
see task in https://github.com/ical-org/ical.net/issues/646